src/arch/src/aarch64/gic/regs/mod.rs

```diff
@@ -2,17 +2,132 @@ mod dist_regs;
 mod icc_regs;
 mod redist_regs;
 
-use crate::aarch64::gic::Result;
-
+use crate::aarch64::gic::{Error, Result};
+use kvm_bindings::kvm_device_attr;
 use kvm_ioctls::DeviceFd;
+use std::iter::StepBy;
+use std::ops::Range;
 use versionize::{VersionMap, Versionize, VersionizeResult};
 use versionize_derive::Versionize;
 
+pub trait MmappedReg {
+    fn range(&self) -> Range<u64>;
+
+    fn iter<T>(&self) -> StepBy<Range<u64>>
+    where
+        Self: Sized,
+    {
+        self.range().step_by(std::mem::size_of::<T>())
+    }
+}
+
+pub trait VgicRegEngine {
```

Comment:
Does this trait need to be `pub`? The trait defines many functions (all public because of the trait) of which only `get/set_regs_data` are the ones to be used "publicly".

As a very simple band-aid we could just make the trait `non-pub` and expect the code _inside_ this crate to properly use the trait methods (with no expectations from code external to this crate).

Reply:
The trait has to be `pub` in order to be used outside of this file. But we don't re-export it outside this module.

Reply:
nit: make these `pub(crate)`